wordpress-mobile/WordPress-Android
# [Modal Layout Picker] Add analytics metrics for the layout picker

**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2454

**Related PRs:** 
Depends on: https://github.com/wordpress-mobile/WordPress-Android/pull/13085

## Description

This PR adds analytics metrics to the layout picker's actions

## To test:

An easy way to test for these analytics is to debug the app and filter Logcat to look for the string `🔵 Tracked:`

Layout Picker should show [when creating a new page from *My Site* or *Site Pages* the *Modal Layout Picker* appears](https://github.com/wordpress-mobile/WordPress-Android/pull/12482).

- Start by navigating to the Modal Layout Picker

### Template Create Page
1. Select a layout
1. Select Create Page
1. In the console, you'll see an event like:
`🔵 Tracked: editor_session_template_apply, Properties: {"has_unsupported_blocks":"0","template":"about-4","editor":"gutenberg","content_type":"new","session_id":"ba5e7e37-a29e-48a8-bbac-7e34625ee5c0","post_type":"page","blog_type":"wpcom","editor_has_hw_disabled":"0"}`

### Template Preview 
1. Select a layout
1. Select Preview
1. In the console, you'll see an event like:
`🔵 Tracked: editor_session_template_preview, Properties: {"has_unsupported_blocks":"0","template":"about-4","editor":"gutenberg","content_type":"new","session_id":"af74365c-abbe-42fc-b969-bafc7c3745b1","post_type":"page","blog_type":"wpcom","editor_has_hw_disabled":"0"}`


### Template Preview, then Create
1. Select a layout
1. Select Preview
1. In the console, you'll see an event like:
`🔵 Tracked: editor_session_template_apply, Properties: {"has_unsupported_blocks":"0","template":"about-4","editor":"gutenberg","content_type":"new","session_id":"ba5e7e37-a29e-48a8-bbac-7e34625ee5c0","post_type":"page","blog_type":"wpcom","editor_has_hw_disabled":"0"}`
1. Repeat selecting back and choosing a new layout as many times as you want 
1. Select Create Page
1. In the console, you'll see an event like:
`🔵 Tracked: editor_session_template_preview, Properties: {"has_unsupported_blocks":"0","template":"about-4","editor":"gutenberg","content_type":"new","session_id":"af74365c-abbe-42fc-b969-bafc7c3745b1","post_type":"page","blog_type":"wpcom","editor_has_hw_disabled":"0"}`

### Validate Editor Session

1. Create a new page from a selected layout using either of the paths mentioned above.
1. Perform some actions in the editor session like publishing, saving as a draft, or switching to the classic editor.
1. Expect to see editor events like the below containing the template field:
`🔵 Tracked: editor_session_start, Properties: {"has_unsupported_blocks":"0","template":"about-4","editor":"gutenberg","content_type":"new","startup_time_ms":7461,"session_id":"ba5e7e37-a29e-48a8-bbac-7e34625ee5c0","post_type":"page","blog_type":"wpcom","editor_has_hw_disabled":"0","unsupported_blocks":[]}`

### Create a Blank Page
1. Create a Blank Page
**Expect** The events mentioned in the last step to not have the template field populated. 

PR submission checklist:
